src/components/earn-opportunity-item/index.tsx

```diff
@@ -0,0 +1,239 @@
+          />
+        )}
+      </View>
+      <View style={styles.mainContent}>
+        <View style={[styles.tokensContainer, styles.row]}>
+          <EarnOpportunityTokens stakeTokens={stakeTokens} rewardToken={rewardToken} />
+          <View>
+            <Text style={styles.aprText}>APR: {formattedApr}%</Text>
+            <View style={styles.earnSource}>
+              {isLiquidityBaking ? (
+                <View style={[styles.earnSourceIcon, styles.liquidityBakingIconWrapper]}>
+                  <Icon size={formatSize(6)} name={IconNameEnum.LiquidityBakingLogo} />
+                </View>
+              ) : (
+                <Icon
+                  style={styles.earnSourceIcon}
+                  size={formatSize(12)}
+                  name={itemIsFarm ? IconNameEnum.QsEarnSource : youvesIconName}
+                />
+              )}
+              <Text style={styles.attributeTitle}>
+                {isLiquidityBaking && 'Liquidity Baking'}
+                {!isLiquidityBaking && itemIsFarm && 'Quipuswap'}
+                {!itemIsFarm && 'Youves'}
+              </Text>
+            </View>
+          </View>
+        </View>
+
+        <HorizontalBorder />
+
+        <Divider size={formatSize(8)} />
+
+        <View style={[styles.row, styles.mb16]}>
+          <StatsItem
+            title="Your deposit:"
+            value={
+              itemIsFarm ? (
+                <FormattedAmountWithLoader
+                  isLoading={stakeIsLoading}
+                  isDollarValue
+                  amount={depositFiatEquivalent}
+                  style={styles.attributeValue}
+                  renderLoader={renderStatsLoader}
+                />
+              ) : (
+                <FormattedAmountWithLoader
+                  isLoading={stakeIsLoading}
+                  symbol={stakedToken.metadata.symbol}
+                  amount={depositAmount}
+                  style={styles.attributeValue}
+                  renderLoader={renderStatsLoader}
+                />
+              )
+            }
+            fiatEquivalent={itemIsFarm ? undefined : depositFiatEquivalent}
+          />
+          {!depositIsZero && !isLiquidityBaking && (
+            <StatsItem
+              title="Claimable rewards:"
+              value={
+                itemIsFarm ? (
+                  <FormattedAmountWithLoader
+                    isLoading={stakeIsLoading}
+                    isDollarValue
+                    amount={claimableRewardsFiatEquivalent}
+                    style={styles.attributeValue}
+                    renderLoader={renderStatsLoader}
+                  />
+                ) : (
+                  <FormattedAmountWithLoader
+                    isLoading={stakeIsLoading}
+                    symbol={rewardToken.symbol}
+                    amount={claimableRewardsAmount}
+                    style={styles.attributeValue}
+                    renderLoader={renderStatsLoader}
```

Comment:
I see that we have similar `isLoading` and `renderLoader` for all occurrences in this file.
And also for `itemIsFarm = true` we have differences only in amount 
Maybe we can somehow move this to another component and simplify this file?

Reply:
Actually, another difference is `isDollarValue` flag